docker_images/sklearn/app/pipelines/tabular_classification.py

```diff
@@ -1,49 +1,129 @@
                 # this is the default path for configuration of a scikit-learn
                 # project. If the project is created using `skops`, it should have
                 # this file.
                 config = json.load(f)
-            model_file = config["sklearn"]["model"]["file"]
-        except Exception as e:
-            print(e)
-            # If for whatever reason we fail to detect requirements of the project,
-            # we install the latest scikit-learn.
-            # TODO: we should log this, or warn or something.
-            model_file = DEFAULT_FILENAME
+        except Exception:
+            # If for whatever reason we fail to detect requirements of the
+            # project, we install the latest scikit-learn.
+            config = dict()
+
+        self.model_file = (
+            config.get("sklearn", {}).get("model", {}).get("file", DEFAULT_FILENAME)
+        )
 
-        self.model = joblib.load(open(Path(cached_folder) / model_file, "rb"))
+        try:
+            with warnings.catch_warnings(record=True) as record:
+                self.model = joblib.load(
+                    open(Path(cached_folder) / self.model_file, "rb")
+                )
+                if len(record) > 0:
+                    # if there's a warning while loading the model, we save it so
+                    # that it can be raised to the user when __call__ is called.
+                    self._load_warnings += record
+        except Exception as e:
+            # if there is an exception while loading the model, we save it to
+            # raise the write error when __call__ is called.
+            self._load_exception = e
+        # use column names from the config file if available, to give the data
+        # to the model in the right order.
+        self.columns = config.get("sklearn", {}).get("columns", None)
 
     def __call__(
         self, inputs: Dict[str, Dict[str, List[Union[str, float]]]]
     ) -> List[Union[str, float]]:
-        """
+        """Return the output of the model for given inputs.
+
         Args:
             inputs (:obj:`dict`):
                 a dictionary containing a key 'data' mapping to a dict in which
                 the values represent each column.
         Return:
-            A :obj:`list` of floats or strings: The classification output for each row.
+            A :obj:`list` of floats or strings: The classification output for
+            each row.
         """
-        column_values = list(inputs["data"].values())
-        rows = list(zip(*column_values))
-        # TODO: we should do sth with the warnings.
-        with warnings.catch_warnings():
-            warnings.filterwarnings("ignore")
-            return self.model.predict(rows).tolist()
+        if getattr(self, "_load_exception", None):
+            # there has been an error while loading the model. We need to raise
+            # that, and can't call predict on the model.
+            raise ValueError(
```

Comment:
cause we need to have it raised every time the user sends an API request so that they get the right error. This is caught and handled correctly in `routes.py`.

Reply:
Great, thanks for explaining!